### PR TITLE
DTFS2-4414 - TFM GEF mapping

### PR DESCRIPTION
**Pull request by ttbarnes**
**Summary**
- TFM mapping for GEF deal
- TFM mapping for GEF facilities
- TFM mapping for GEF facilities in a deal
- TFM mapping for GEF deals combined with BSS/EWCS deals

**Refactoring/improvements**
- refactor some deal/facility field mapping functions to be more generic and not tied to BSS/EWCS specific fields 
- split up deals reducer, functional
- improve/add test coverage for deals and facility reducers

**Template improvements for GEF variation in TFM**
- add ‘dash if empty’ logic to various templates
- fix some template issues where things like currency code or percentage symbols would render even if there is no value

**Note**
This won't be able to be tested by QA as it's new mappings for GEF deals and facilities manually added to the DB. Would be great if PR reviewer could add these deals to DB and make sure TFM loads.

### CODE REVIEW

File: trade-finance-manager-ui/templates/case/parties/_macros/parties-key-value-grid-row.njk [hunk 1]
```diff
@@ -9,10 +9,14 @@
         {% endif %}
     </div>
     <div class="govuk-grid-column-two-thirds">
-      {% if params.dataCy | safe %}<p class="govuk-body-s govuk-!-margin-top-3 {{ params.valueStyle }}" data-cy="{{ params.dataCy }}">{% else %}<p class="govuk-body-s {{ params.valueStyle }}">{% endif %}
-        {{ params.value }}
+      {% if params.dataCy | safe %}
+        <p class="govuk-body-s govuk-!-margin-top-3 {{ params.valueStyle }}" data-cy="{{ params.dataCy }}">
```
Comment (pricea1) [inline trade-finance-manager-ui/templates/case/parties/_macros/parties-key-value-grid-row.njk:13]: Why is there a top margin if dataCy exists?

**Review by pricea1 — APPROVED** [review]
phew, got there in the end.
A couple of comments but otherwise good to go

File: trade-finance-manager-ui/templates/case/parties/_macros/parties-key-value-grid-row.njk (hunk 1 above)
Comment (ttbarnes) [reply trade-finance-manager-ui/templates/case/parties/_macros/parties-key-value-grid-row.njk:13]: Honestly I don't know, the macros in this section of TFM are not very clear. I didn't add or modify this part, I just changed the indentation so I could read it and add a "dashIfEmpty" nunjucks filter 🙃

File: trade-finance-manager-api/src/graphql/reducers/mappings/gef-facilities/mapGefFacilities.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const mapGefFacility = require('./mapGefFacility');
+
+const mapGefFacilities = (dealSnapshot) => {
+  const { facilities } = dealSnapshot;
+
+  const mappedFacilities = [];
+
+  facilities.forEach((f) => {
```
[the remark this answers is not in the document]
Comment (ttbarnes) [reply trade-finance-manager-api/src/graphql/reducers/mappings/gef-facilities/mapGefFacilities.js]: Good stuff, i've simplified this

**Comment by ttbarnes** [comment]
Thank you so much for reviewing this beast of a PR

**Pull request merged**